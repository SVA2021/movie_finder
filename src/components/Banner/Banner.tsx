import CloseIcon from '@mui/icons-material/Close';
import {Box, Typography, useTheme} from "@mui/material";
import {FC} from "react";
import {StyledButton} from "..";

interface BannerProps {
  closeHandler: () => void
}

export const Banner: FC<BannerProps> = ({closeHandler}) => {

  const theme = useTheme();
  const sva = 'sva'.split('');

  const handleClick = (e: React.MouseEvent) => {
    e.preventDefault();
    closeHandler();
  }

  return (
    <Box
      component={'a'}
      href={'https://github.com/SVA2021/movie_finder'}
      target={'_blank'}
      rel={'noopener noreferref'}
      sx={{
        mt: 2,
        mb: 2,
        position: 'relative',
        height: '100px',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        color: 'common.white',
        textDecoration: 'none',
        cursor: 'pointer',
        borderRadius: 4,
        "&:hover": {
          boxShadow: `8px 8px 8px ${theme.palette.action.hover},
					-8px -8px 8px ${theme.palette.action.hover}`,
        },
      }}
    >
      {
        sva.map((letter) =>
          <Typography variant="caption" component={'span'} key={letter}
            sx={{
              m: 2,
              transition: 'all 0.2s ease-in-out',
              textTransform: 'uppercase',
              textAlign: 'center',
              color: 'transparent',
              WebkitTextStrokeWidth: '2px',
              WebkitTextStrokeColor: theme.palette.action.focus,
              '&:hover': {
                WebkitTextStrokeColor: theme.palette.action.hover,
              },
            }}
          >
            {letter}
          </Typography>
        )
      }

      <Box sx={{position: 'absolute', top: 4, right: 4}}>
        <StyledButton onClick={handleClick} ><CloseIcon fontSize='small' /></StyledButton>
      </Box>

    </Box>
  )
}