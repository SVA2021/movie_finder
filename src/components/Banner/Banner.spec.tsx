import {render, screen, fireEvent} from "@testing-library/react";
import {Banner, myPortfolioLink} from "./Banner";

describe('render Banner', () => {

  const onClick = jest.fn();
  const setUp = () => render(<Banner closeHandler={onClick} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('banner')).toMatchSnapshot();
  });

  it('check s', () => {
    setUp();
    expect(screen.getByText(/s/i)).toBeInTheDocument();
  });

  it('check v', () => {
    setUp();
    expect(screen.getByText(/v/i)).toBeInTheDocument();
  });

  it('check a', () => {
    setUp();
    expect(screen.getByText(/a/i)).toBeInTheDocument();
  });
  
  it('check link href', () => {
    setUp();
    expect(screen.getByRole('link')).toHaveAttribute('href', myPortfolioLink)
  });

  it('check button', () => {
    setUp();
    fireEvent.click(screen.getByRole('button'))
    expect(onClick).toBeCalled();
  });

})