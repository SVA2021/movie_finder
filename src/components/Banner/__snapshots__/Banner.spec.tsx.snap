// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Banner snapshot test 1`] = `
<a
  class="MuiBox-root css-pw2lgz"
  data-testid="banner"
  href="https://portfolio-sva2021.vercel.app"
  rel="noopener noreferref"
  target="_blank"
>
  <span
    class="MuiTypography-root MuiTypography-caption css-80zbff-MuiTypography-root"
  >
    s
  </span>
  <span
    class="MuiTypography-root MuiTypography-caption css-80zbff-MuiTypography-root"
  >
    v
  </span>
  <span
    class="MuiTypography-root MuiTypography-caption css-80zbff-MuiTypography-root"
  >
    a
  </span>
  <div
    class="MuiBox-root css-bymsth"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1nsf0ti-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</a>
`;
